x-pack/plugins/data_enhanced/server/search/es_search_strategy.ts

```diff
@@ -33,15 +33,19 @@ import {
 } from './request_utils';
 import { toAsyncKibanaSearchResponse } from './response_utils';
 import { AsyncSearchResponse } from './types';
-import { KbnServerError } from '../../../../../src/plugins/kibana_utils/server';
+import { getKbnServerError, KbnServerError } from '../../../../../src/plugins/kibana_utils/server';
 
 export const enhancedEsSearchStrategyProvider = (
   config$: Observable<SharedGlobalConfig>,
   logger: Logger,
   usage?: SearchUsage
 ): ISearchStrategy<IEsSearchRequest> => {
   async function cancelAsyncSearch(id: string, esClient: IScopedClusterClient) {
-    await esClient.asCurrentUser.asyncSearch.delete({ id });
+    try {
+      await esClient.asCurrentUser.asyncSearch.delete({ id });
+    } catch (e) {
+      throw getKbnServerError(e);
```

Comment:
Is it worth updating the other search strategies, or at least letting them know that this is the preferred way to handle errors?